docs/proposals/20231122-automate-aks-features.md

```diff
@@ -0,0 +1,308 @@
+adopting existing AKS clusters into CAPZ
+(https://github.com/kubernetes-sigs/cluster-api-provider-azure/issues/1173) additionally would only require
+the extra steps to create the AzureManagedControlPlane referring to the adopted ManagedCluster.
+
+One other consideration is that this approach trades the requirement that CAPZ has the necessary RBAC
+permissions to create ManagedClusters for users having that same capability. CAPZ would still require
+permissions to read, update, and delete ManagedClusters.
+
+The main roadblocks with this method relate to ClusterClass. If CAPZ's AzureManagedControlPlane controller is
+not responsible for creating the ASO ManagedCluster resource, then users would need to manage those
+separately, defeating much of the purpose of ClusterClass. Additionally, since each AzureManagedControlPlane
+will be referring to a distinct ManagedCluster, the new `ManagedClusterName` field should not be defined in an
+AzureManagedControlPlaneTemplate.
+
+#### Option 2: CAPZ resource references a non-functional ASO "template" resource
+
+This method is similar to [Option 1]. To better enable ClusterClass, instead of defining the full name of the
+ManagedCluster resource, the name of a "template" resource is defined instead:
+
+```go
+type AzureManagedControlPlaneClassSpec struct {
+	...
+	
+	// ManagedClusterTemplateName is the name of the ASO ManagedCluster to be used as a template from which
+	// new ManagedClusters will be created.
+	ManagedClusterTemplateName string `json:"managedClusterTemplateName"`
+}
+```
+
+This template resource will be a ManagedCluster used as a base from which the AzureManagedControlPlane
+controller will create a new ManagedCluster. The template ManagedCluster will have ASO's [`skip` reconcile
+policy](https://azure.github.io/azure-service-operator/guide/annotations/#serviceoperatorazurecomreconcile-policy)
+applied so it does not result in any AKS resource being created in Azure. The ManagedClusters created based on
+the template will be reconciled normally to create AKS resources in Azure. The non-template ManagedClusters
+will be linked to the AzureManagedControlPlane through the standard `cluster.x-k8s.io/cluster-name` label.
+
+To modify parameters on the AKS cluster, either the template or non-template ManagedCluster may be updated.
+CAPZ will propagate changes made to a template to instances of that template. Parameters defined on the
+template take precedence over the same parameters on the instances.
+
+The main difference with [Option 1] that enables ClusterClass is that the same ManagedCluster template
+resource can be referenced by multiple AzureManagedControlPlanes, so this new `ManagedClusterTemplateName`
+field can be defined on the AzureManagedControlPlaneClassSpec so a set of ASK parameters defined once in the
+template can be applied to all Clusters built from that ClusterClass.
```

Comment:
I like this approach!!

Reply:
One thing I fear with this is that the pattern is modeled roughly after how ClusterClass works between InfraClusterTemplate and InfraCluster resources, but isn't exactly the same. That weird spot of being almost-but-not-quite-the-same might be problematic.